### PR TITLE
add error handler middleware

### PR DESCRIPTION
**Pull request by rajeman**
#### What does this PR do?

It ensures an error middleware is available for failed validation and other errors

#### Description of Task to be completed?

- [x] setup an error handler to be called when validation fails

#### How should this be manually tested?
- whenever an error is caught in any middleware, call next and pass the error object as argument i.e. <code>next(error)</code> 
- whenever a user input fails validation, create a new error object and pass the error message as string i.e. <code>next(new Error('Invalid username or password'))</code>
- In the two cases above, you should see the error description in the response body of the server.

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#163085851](https://www.pivotaltracker.com/n/projects/2229853/stories/163085851).

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ app.use(
 );
 
 app.use('/api/v1/user', userRoutes);
+app.use(errorHandler);
```
Comment (oreoluwade) [inline app.js:31]: This should be the last thing mounted on the app.

hence, should be after the all routes handler.

File: middlewares/errorHandler.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line no-unused-vars
+export default (err, req, res, next) => {
+  res.status(err.status || 500).send({
+    errors: {
```
Comment (oreoluwade) [inline middlewares/errorHandler.js:4]: Your error response does not match what was given:

`{
  "errors":{
    "body": [
      "can't be empty"
    ]
  }
}`

Comment (oreoluwade) [reply middlewares/errorHandler.js:4]: Check the error section in the README for clarification

File: app.js (hunk 1 above)
Comment (rajeman) [reply app.js:31]: I will do that now.

File: middlewares/errorHandler.js (hunk 2 above)
Comment (rajeman) [reply middlewares/errorHandler.js:4]: I will do that now

Comment (rajeman) [reply middlewares/errorHandler.js:4]: > Check the error section in the README for clarification

I have made the changes

**Comment by oreoluwade** [comment]
@rajeman Please help remove the all route handler from app.js and all references to it as well. Want to merge this asap

**Comment by rajeman** [comment]
> @rajeman Please help remove the all route handler from app.js and all references to it as well. Want to merge this asap

Alright

**Comment by rajeman** [comment]
> @rajeman Please help remove the all route handler from app.js and all references to it as well. Want to merge this asap

I have removed it

**Review by oreoluwade — APPROVED** [review]
Good work!

**Pull request merged**